rpcs3/rpcs3qt/save_manager_dialog.cpp

```diff
@@ -249,36 +256,118 @@ void save_manager_dialog::OnEntryRemove()
+
+		// I could make a QPair, and remove both simultaneously, slightly more efficiently.  
+		// However, the code becomes messier to read IMO with a custom sort comparator and accessing the qpair.
+		QList<int> rows;
+		QSet<int> save_entries;
+		for (const QModelIndex& index : selection.indexes())
+		{
+			rows.append(index.row());
+			int idx_real = m_list->item(index.row(), 0)->data(Qt::UserRole).toInt();
+			save_entries.insert(idx_real);
+		}
+
+		// Remove from UI
+		qSort(rows);
+
+		int prev = -1;
+		for (int i = rows.count() - 1; i >= 0; i -= 1)
+		{
+			int curr = rows[i];
+
+			// Duplicates exist because of the columns also being selected.
+			if (curr != prev)
+			{
+				m_list->removeRow(curr);
+				prev = curr;
+			}
+		}
+
+		// Remove from file system
+		for (int idx_real : save_entries)
+		{
+			fs::remove_all(m_dir + m_save_entries[idx_real].dirName + "/");
+		}
+	}
+}
+
 //Pop-up a small context-menu, being a replacement for save_data_manage_dialog
 void save_manager_dialog::ShowContextMenu(const QPoint &pos)
 {
+	bool selectedItems = m_list->selectedItems().size() > 4; // > 4 is slight hack due to four columns-- more than one row selected.
+
 	QPoint globalPos = m_list->mapToGlobal(pos);
 	QMenu* menu = new QMenu();
 	int idx = m_list->currentRow();
+	if (idx == -1)
+	{
+		return;
+	}
 
 	QAction* saveIDAct = new QAction(tr("SaveID"), this);
 	QAction* titleAct = new QAction(tr("Title"), this);
 	QAction* subtitleAct = new QAction(tr("Subtitle"), this);
+
 	QAction* removeAct = new QAction(tr("&Remove"), this);
 	QAction* infoAct = new QAction(tr("&Info"), this);
+	QAction* showDirAct = new QAction(tr("&Open Save Directory"), this);
 
-	//This is also a stub for the sort setting. Ids is set according to their sort-type integer.
-	m_sort_options = new QMenu(tr("&Sort"));
+	//This is also a stub for the sort setting. Ids are set accordingly to their sort-type integer.
+	// TODO: add more sorting types.
+	m_sort_options = new QMenu(tr("&Sort By"));
 	m_sort_options->addAction(titleAct);
 	m_sort_options->addAction(subtitleAct);
 	m_sort_options->addAction(saveIDAct);
 
 	menu->addMenu(m_sort_options);
 	menu->addSeparator();
 	menu->addAction(removeAct);
-	menu->addSeparator();
 	menu->addAction(infoAct);
+	menu->addAction(showDirAct);
 
+	infoAct->setEnabled(!selectedItems);
+	showDirAct->setEnabled(!selectedItems);
 	removeAct->setEnabled(idx != -1);
 
-	//Events
-	connect(removeAct, &QAction::triggered, this, &save_manager_dialog::OnEntryRemove);
+	// Events
+	connect(removeAct, &QAction::triggered, this, &save_manager_dialog::OnEntriesRemove); // entriesremove handles case of one as well
 	connect(infoAct, &QAction::triggered, this, &save_manager_dialog::OnEntryInfo);
+	connect(showDirAct, &QAction::triggered, [=]() {
+		int idx_real = m_list->item(idx, 0)->data(Qt::UserRole).toInt();
+		QString path = qstr(m_dir + m_save_entries[idx_real].dirName + "/");
+
+		QProcess* process = new QProcess();
```

Comment:
This leaks. You can just use `QProcess::startDetached(command, QStringList(params))` instead. Also, if you only want to show specific dir (without highlighting specific file) [openURL](http://doc.qt.io/qt-5.9/qdesktopservices.html#openUrl) should work in a cross-platform way.

Reply:
Thanks. I fixed the other use of improper use of QProcess as well with this information.